source/common/http/http2/metadata_encoder.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/logger.h"
+#include "common/http/http2/metadata_interface.h"
+
+namespace Envoy {
+namespace Http {
+namespace Http2 {
+
+/**
+ * A class that creates and sends METADATA payload. The METADATA payload is a group of string key
+ * value pairs encoded in HTTP/2 header blocks.
+ */
+class MetadataEncoder : Logger::Loggable<Logger::Id::metadata> {
+public:
+  MetadataEncoder(uint64_t stream_id) : stream_id_(stream_id) {}
+
+  /**
+   * Creates wire format HTTP/2 header block from metadata_map. Only after previous payload is
+   * drained, new payload can be encoded. This limitation is caused by nghttp2 can't incrementally
+   * encode header block.
+   * @param metadata_map supplies METADATA to encode.
+   * @return whether encoding is successful.
+   */
+  bool createPayload(MetadataMap& metadata_map);
+
+  /**
+   * Called in nghttp2 callback function after METADATA is submitted. The function releases len data
+   * in payload_.
+   * @param len supplies the size to be released.
+   */
+  void releasePayload(uint64_t len);
+
+  /**
+   * @return true if there is payload to be submitted.
+   */
+  bool hasNextFrame();
+
+  /**
+   * Submit METADATA frame to nghttp2.
+   * @return whether the operation succeeds.
+   */
+  bool submitMetadata() {
+    // TODO(soya3129): Implement this in MetadataHandler and update comments.
```

Comment:
Can we just delete this for now and reintroduce later when it actually does something?

Reply:
Sure! Deleted.